tests/unit/kvmagent/test_vm_plugin_handlers.py

```
@@ -1778,6 +1778,149 @@ def test_reload_redirect_usb(self):
+        rm_file_force.assert_not_called()
+
+    def test_hot_unplug_dgpu_shmem_cleans_file_for_stopped_vm(self, monkeypatch):
+        plugin = _make_vm_plugin()
+        vm = MagicMock(state=vm_plugin.Vm.VM_STATE_SHUTDOWN)
+        rm_file_force = MagicMock()
+        monkeypatch.setattr(vm_plugin, 'get_vm_by_uuid_no_retry', MagicMock(return_value=vm))
+        monkeypatch.setattr(vm_plugin.linux, 'rm_file_force', rm_file_force)
+
+        req = _make_req({
+            'vmUuid': 'vm-uuid',
+            'shmem': {'path': '/dev/shm/tf_vm-uuid', 'size': 2 * 1024 * 1024},
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/dev/shm/tf_vm-uuid"

(S108)